flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -131,6 +134,8 @@
 	/** Lazily initialized list of files to ship */
 	protected List<File> shipFiles = new LinkedList<>();
 
+	private List<URI> archives = new LinkedList<>();
```

Comment:
`LinkedLists` are not so fast. Better to use an `ArrayList` and initialize it with an initial capacity.